### ISSUE TITLE
Please Fix - Sonoma Support, Fix or Workaround?

### ISSUE THREAD

**Issue by paulmoons** [packagesdev/savehollywood#24]
All the workarounds online no longer work for Sonoma. I suspect it's a small tweak needed as the preview thumbnail actually shows the videos but when the screen saver is engaged fullscreen it just results in "No videos." 

Can someone please take over this project as there is nothing else out there - I'm happy to help if I can!

**Comment by paulmoons** [comment]
Holy, erm.. shit, I got it working! Just drop your videos into: 

Macintosh HD > Library > Screen Savers 

Note that you can't use a subfolder and also for some reason the _fit screen_ option doesn't seem to work anymore.. but otherwise Save Hollywood seems to be working again! 

Hooray for trial and error!

**Comment by tsuftko** [comment]
Hey paulmoons.... did the same thing HD>Library>Screen Savers, no go though.  Preview pane is showing the vid, when initiated I still get "No Videos".  Was there an extra step or something you did when loading the vid into the options menu for SH?

**Comment by paulmoons** [comment]
I'm not sure if it's related but I had some weird issues with the settings not working at first - fill vs fit screen. As a fix I did the following:

1. Uninstalled SaveHollywood (I kept my videos in place)
2. Selected a different screensaver and made sure it works.
3. Restarted
4. Installed SaveHollywood again and selected it as my screen saver.
 
Everything worked after that - including settings. Beyond that, just make sure:

- Rosetta 2 from Apple is installed.
- The videos you are using are compatible.
- You don't nest the videos - The preview will work, but unless they're directly in the HD > Library > Screensaver folder, SaveHollywood doesn't seem to work properly.

It definitely works though - I'm running MacOS Sonoma 14.2.1 on an M3 MacBook Pro FYI and the latest available download of SaveHollywood.

If I knew how I'd recompile it for Apple Silicon and apply whatever fixes necessary as it's a great little screensaver.

Best of luck with it!

**Comment by MDanihy** [comment]
This didn't work for me.  For one, it has to be installed for all users, if you install it just for yourself, Options doesn't work.  Dumping the videos in the same directory didn't work as well..

Sonoma 14.3 with M3 iMac

**Comment by paulmoons** [comment]
> This didn't work for me. For one, it has to be installed for all users, if you install it just for yourself, Options doesn't work. Dumping the videos in the same directory didn't work as well..
> 
> Sonoma 14.3 with M3 iMac

I'm pretty sure I install it for all users.. and following the instructions I posted above Save Hollywood does indeed work - I have since upgraded to MacOS 14.3 and it continues to work as expected. The only issue is that sometimes the legacy screensaver task will remain in memory eating processor time but that's a Sonoma bug from what I understand.

**Comment by Krazubu** [comment]
I reinstalled 2.5, tried to put into /Library/Screen Savers, this didn't work.
I logged out and back into my session, and voilà it worked again. No need to put inside /Library/Screen Savers
I suspect some sandboxing / authorisation is fixed by the reinstall, but it still needs a relog or restart.

**Comment by MDanihy** [comment]
What worked for me was to put the videos in a folder within /Users/Shared and they worked fine.  The caveat here is that none of the options worked.  The audio still played and the resolution of the videos didn't randomize or listen to any of the video size settings.

**Comment by paulmoons** [comment]
It's clearly on the edge of breaking for good. Hopefully someone can pick-up the code as I haven't been able to find anything out there that can effectively replicate this screensaver.

**Comment by MDanihy** [comment]
I believe the issue is Apple has not released any APIs for native screen savers.  Everything has to go through the legacy screen saver system, which has a tendency to keep running in the background using up resources while you work.  If the APIs are released, I am sure it will be a while before we see a replacement.

**Comment by franklin-like-the-turtle** [comment]
+1 For the /Users/Shared approach from @MDanihy.

Once again [Daredevil Dan](https://www.youtube.com/watch?v=1_rRTJu6yk0) graces the screen of my M1 pro...uh...work MacBook.

Sonoma 14.7.2

**Comment by TomiGio** [comment]
+2 for @MDanihy 's approach.
works perfectly un Sequoia 15.5

**Comment by paulmoons** [comment]
Wish there was an Apple Silicon version/alternative as support is going to end eventually. Ugh.

**Comment by rauldsl** [comment]
hey folks @TomiGio @MDanihy 
I have a Mac with Apple M1 running macOS Sequoia 15.5, and SaveHollywood is not working.

I tried placing the .mp4 video file in /Users/Shared and also in Macintosh HD/Library/Screen Savers/, but nothing happens. I even reinstalled SaveHollywood for all users, but it’s still not working — the video doesn’t play.

Just a black screen in preview mode.

Any thoughts?

**Comment by paulmoons** [comment]
@rauldsl I'm running Sequoia 15.6 and it seems to work.. just. It works erratically when an external monitor is connected and can sometimes stop working other times but it generally works. My videos are stored in Macintosh HD/Library/Screen Savers/ I get the impression it's on its last legs and without some work, it is only a matter of time before it stops working altogether.

**Comment by HansSAQ** [comment]
The logout/login worked for me.  I also had to go find my video again (somehow lost from the settings)

**Comment by balkangirl67** [comment]
Heyy, I have downloaded savehollywood and everything worked just fine. I had to drag the video into the users/shared/library/application support so that the video would appear on the screensaver option. But now I have a different problem. The sound from the video won't go away after the screensaver ends. It just keeps going. I need to always restart my MacBook every time the screensaver ends so that the sound would go away. How do I fix this? I can also hear an echo in the audio sometimes.

**Comment by crose2029-oss** [comment]
> [@rauldsl](https://github.com/rauldsl) I'm running Sequoia 15.6 and it seems to work.. just. It works erratically when an external monitor is connected and can sometimes stop working other times but it generally works. My videos are stored in Macintosh HD/Library/Screen Savers/ I get the impression it's on its last legs and without some work, it is only a matter of time before it stops working altogether.

hi, my problem is similar but a bit diffferent. who i click the SaveHollywood screensaver in settings, it does nothing. the preview works, but how do i enable it? im on the same update as you. it doesnt have the same options as other screensavers where it says "show as wallpaper" and i dont even get the no videos screen. it simply wont select and stays as my pevious wallpaper

**Comment by miikememe** [comment]
@balkangirl67 while i don't have a real solution yet, you don't have to restart, you just need to quit/force quit the legacy screensaver process from activity monitor.. could also setup a script to do it with one click.

**Comment by pvinis** [comment]
Indeed, it finally works again for macOS 26 and 27 and Apple Silicon! https://github.com/pvinis/save-bollywood, `brew install pvinis/pvinis/save-bollywood`. Enjoy!

**Comment by paulmoons** [comment]
> Indeed, it finally works again for macOS 26 and 27 and Apple Silicon! https://github.com/pvinis/save-bollywood, `brew install pvinis/pvinis/save-bollywood`. Enjoy!

LEGEND!
